From 3ee63f17cdcfec7380cc3f3b4335d58c6dddbf24 Mon Sep 17 00:00:00 2001
From: Billy Stevens <contact@wasv.me>
Date: Sat, 3 Jul 2021 01:29:52 -0400
Subject: [PATCH] Removes 'using namespace std'.

Was not needed, and caused build errors with latest GCC.

Fixes matrix-io/xc3sprog#29.

Upstream-Status: Pending
---
 bitfile.cpp  | 6 ++----
 srecfile.cpp | 2 --
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/bitfile.cpp b/bitfile.cpp
index 327d86e035..6a402bb1f3 100644
--- a/bitfile.cpp
+++ b/bitfile.cpp
@@ -35,8 +35,6 @@ Dmitry Teytelman [dimtey@gmail.com] 14 Jun 2006 [applied 13 Aug 2006]:
 #include <time.h>
 #include "bitrev.h"
 
-using namespace std;
-
 BitFile::BitFile()
   : length(0)
   , buffer(0)
@@ -607,14 +605,14 @@ uint32_t BitFile::saveAs(FILE_STYLE style, const char  *device,
   return clip;
 }
 
-void BitFile::error(const string &str)
+void BitFile::error(const std::string &str)
 {
   errorStr=str;
   Error=true;
   fprintf(logfile,"%s\n",str.c_str());
 }
 
-void BitFile::readField(string &field, FILE *fp)
+void BitFile::readField(std::string &field, FILE *fp)
 {
   byte t[2];
   fread(t,1,2,fp);
diff --git a/srecfile.cpp b/srecfile.cpp
index 2fff847445..5f421063fa 100644
--- a/srecfile.cpp
+++ b/srecfile.cpp
@@ -28,8 +28,6 @@ Modifyied from srecdec
 #include <string.h>
 #include <stdlib.h>
 
-using namespace std;
-
 int SrecFile::DecodeSRecordLine(char *source, unsigned char *dest, S_Record *SRec)
 {
   char buffer[16];
